e2e/scripts/st_arrow_dataframe_config.py

```diff
@@ -30,9 +31,398 @@
+            validate="^[0-9]+$",  # Should be ignored
+        ),
+        "col_1": st.column_config.TextColumn(),
+    },
+)
+
+st.header("Number column:")
+st.dataframe(
+    pd.DataFrame(
+        {
+            "col_0": [1, 2, 3, None],
+            "col_1": ["1", "2", "invalid", None],
+        }
+    ),
+    column_config={
+        "col_0": st.column_config.NumberColumn(
+            "Number column",
+            width="medium",
+            help="This is a number column",
+            required=True,  # Should be ignored
+            disabled=False,  # Should be ignored
+            default=0,  # Should be ignored
+            min_value=5,  # Should be ignored
+            max_value=10,  # Should be ignored
+            step=0.001,
+        ),
+        "col_1": st.column_config.NumberColumn(
+            format="%.2f%%",
+        ),
+    },
+)
+
+st.header("Checkbox column:")
+st.dataframe(
+    pd.DataFrame(
+        {
+            "col_0": [True, False, False, None],
+            "col_1": ["yes", "no", "invalid", None],
+        }
+    ),
+    column_config={
+        "col_0": st.column_config.CheckboxColumn(
+            "Checkbox column",
+            width="medium",
+            help="This is a checkbox column",
+            required=True,  # Should be ignored
+            disabled=False,  # Should be ignored
+            default=True,  # Should be ignored
+        ),
+        "col_1": st.column_config.CheckboxColumn(),
+    },
+)
+
+st.header("Selectbox column:")
+
+st.dataframe(
+    pd.DataFrame(
+        {
+            "col_0": [1, 2, 3, None],
+            "col_1": ["a", "b", "c", None],
+        }
+    ),
+    column_config={
+        "col_0": st.column_config.SelectboxColumn(
+            "Selectbox column",
+            width="medium",
+            help="This is a selectbox column",
+            required=True,  # Should be ignored
+            disabled=False,  # Should be ignored
+            default=True,  # Should be ignored
+            options=[1, 2, 3, 4, 5],
+        ),
+        "col_1": st.column_config.SelectboxColumn(),
+    },
+)
+
+st.header("Link column:")
+
+st.dataframe(
+    pd.DataFrame(
+        {
+            "col_0": [
+                "https://streamlit.io/",
+                "https://docs.streamlit.io/",
+                "https://streamlit.io/gallery",
+                None,
+            ],
+            "col_1": ["/a", "/b", "", None],
+        }
+    ),
+    column_config={
+        "col_0": st.column_config.LinkColumn(
+            "Link column",
+            width="medium",
+            help="This is a link column",
+            required=True,  # Should be ignored
+            disabled=False,  # Should be ignored
+            default="https://streamlit.io/",  # Should be ignored
+            max_chars=5,  # Should be ignored
+            validate="^[0-9]+$",  # Should be ignored
+        ),
+        "col_1": st.column_config.LinkColumn(),
+    },
+)
+
+st.header("Datetime column:")
```

Comment:
I know we have a lot of tests but I think it still might be worth to add some e2e tests checking for `Datetime -> Time` and `Datetime -> Date`.

Reply:
Added a few columns to tests these scenarios 👍

Reply:
I assume you meant something like: a column with datetime values rendered via the `date` column type, or?

Reply:
Yep!